agent/src/util/device_plugin_service.rs

```diff
@@ -360,25 +474,286 @@ impl InstanceDevicePlugin {
+
+                trace!(
+                    "InstanceDevicePlugin::allocate - finished processing device_usage_id {}",
+                    device_usage_id
+                );
+            }
+            // Successfully reserved device_usage_slot[s] for this node.
+            // Add response to list of responses
+            let broker_properties =
+                get_all_broker_properties(&dps.config.broker_properties, &akri_device_properties);
+            let response = build_container_allocate_response(
+                broker_properties,
+                akri_annotations,
+                &akri_devices.into_values().collect(),
+            );
+            container_responses.push(response);
+        }
+
+        // Notify device usage had been changed
+        if let Some(sender) = &dps.instance_map.read().await.usage_update_message_sender {
+            if let Err(e) = sender.send(ListAndWatchMessageKind::Continue) {
+                error!(
+                    "InstanceDevicePlugin::allocate - fails to notify device usage, error {}",
+                    e
+                );
+            }
+        }
+        trace!(
+            "InstanceDevicePlugin::allocate - for Instance {} returning responses",
+            &dps.instance_name
+        );
+        Ok(Response::new(v1beta1::AllocateResponse {
+            container_responses,
+        }))
+    }
+}
+
+#[derive(Clone)]
+pub struct ConfigurationDevicePlugin {}
+
+impl ConfigurationDevicePlugin {
+    async fn list_and_watch(
+        &self,
+        dps: Arc<DevicePluginService>,
+        kube_interface: Arc<impl KubeInterface>,
+        kubelet_update_sender: mpsc::Sender<Result<ListAndWatchResponse, Status>>,
+        polling_interval_secs: u64,
+    ) {
+        let mut list_and_watch_message_receiver = dps.list_and_watch_message_sender.subscribe();
+        let mut keep_looping = true;
+        let mut prev_virtual_devices = HashMap::new();
+        while keep_looping {
+            trace!(
+                "ConfigurationDevicePlugin::list_and_watch - loop iteration for Instance {}",
+                dps.instance_name
+            );
+            let instance_map_snapshot = dps.instance_map.read().await.clone();
+            let mut discovered_devices = HashMap::new();
+            for (instance_name, _) in instance_map_snapshot.instances {
+                let virtual_devices = build_list_and_watch_response(
+                    &instance_name,
+                    dps.clone(),
+                    kube_interface.clone(),
+                    |device_usage_id, configuration_usage_slots| {
+                        // allow reallocate if reserved by Configuration
+                        configuration_usage_slots.contains(device_usage_id)
+                    },
+                )
+                .await
+                .unwrap();
+                discovered_devices.insert(instance_name, virtual_devices);
+            }
+            // construct virtual device info list
```

Comment:
nit: consistency with comments (Capitalize the first letter of the comment)

Reply:
update comments